pep-0639.rst

```diff
@@ -1,543 +1,2221 @@
+
+**Tool** *(Sub: Packaging Tool, Build Tool, Install Tool, Publishing Tool)*
+  A program, script or service executed by the user or automatically that
+  seeks to conform to the specification defined in this PEP.
+
+  A **packaging tool** refers to a tool used to build, publish,
+  install, or otherwise directly interact with Python packages.
+
+  A **build tool** is a packaging tool used to generate a source or built
+  distribution from a project source tree or sdist, when directly invoked
+  as such (as opposed to by end-user-facing install tools).
+  Examples: Wheel project, PEP 517 backends via ``build`` or other
+  package-developer-facing frontends, calling ``setup.py`` directly.
+
+  An **install tool** is a packaging tool used to install a source or built
+  distribution in a target environment. Examples include the PyPA pip and
+  ``installer`` projects.
+
+  A **publishing tool** is a packaging tool used to upload distribution
+  archives to a package index, such as Twine for PyPI.
+
+**Wheel Format** *(Short: wheel, Rel: Wheel project)*
+  Here, **wheel**, the standard built distribution format introduced in PEP 427
+  and `specified by PyPA <#wheelspec_>`_, will be referred to in lowercase,
+  while the `Wheel project <#wheelproject_>`_, its reference implementation,
+  will be referred to as **Wheel** in Title Case.
+
+
+Specification
+=============
+
+The changes necessary to implement the improved license handling outlined in
+this PEP include those in both
+`distribution package metadata <Core metadata_>`_, as defined in the
+`core metadata specification <#coremetadataspec_>`_, and
+`author-provided project source metadata <Project source metadata_>`_, as
+originally defined in PEP 621.
+
+Further, `minor additions <License files in project formats_>`_ to the
+source distribution (sdist), built distribution (wheel) and installed project
+specifications will help document and clarify the already allowed,
+now formally standardized behavior in these respects.
+Finally, `guidance is established <Converting legacy metadata_>`_
+for tools handling and converting legacy license metadata to license
+expressions, to ensure the results are consistent, correct and unambiguous.
+
+Note that the guidance on errors and warnings is for tools' default behavior;
+they MAY operate more strictly if users explicitly configure them to do so,
+such as by a CLI flag or a configuration option.
+
+
+Core metadata
+-------------
+
+The `PyPA Core Metadata specification <#coremetadataspec_>`_ defines the names
+and semantics of each of the supported fields in the distribution metadata of
+Python distribution packages and installed projects.
+
+This PEP `adds <Add License-Expression field_>`_ the
+``License-Expression`` field,
+`adds <Add License-File field_>`_ the ``License-File`` field,
+`deprecates <Deprecate License field_>`_ the ``License`` field,
+and `deprecates <Deprecate license classifiers_>`_ the license classifiers
```

Comment:
Usability note: as much as possible, link texts should be different and self-standing.  Here there are multiple links like `adds` or `deprecates`, which are not informative on their own.

(Sometimes we have to use reST substitution markup to have code markup inside link text — not the most convenient thing but it works! https://github.com/python/cpython/blob/main/Doc/library/functions.rst#L43)

Reply:
Yeah here and a few other places I linked less-optimal words to avoid having to resort to such awful hacks, due to the layer of indirection around link/reference targets such that the link text isn't actually being used as the target name (thus voiding the immediate need to be unique), and due to the parallelism in the structure (in the source and in the rendered output) that makes it rather unambiguous what is being referred to. But if this is really a serious blocker sufficient to justify trying to hack around this various places (given how many PEPs don't even use inline links at all), I can give it a shot...